### PR TITLE
Extend the message to explain what is happening

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #16238

### Summary of Changes

In some broken hosting envoirments both mime checker tools are not enabled. In that case show that message with more extended info.

### Testing Instructions

- Try this on a broken hosting.
- confirm the extended message

- review the text.

### Expected result

Extended text explaining what is happening

### Actual result

Just a general message which is not longer match the usage.

### Documentation Changes Required

None

### CODE REVIEW

**Comment by mbabker** [comment]
This needs to be crafted carefully.  This message might be appropriate for the site owner who would be the primary POC regarding server configurations, but for a general user of the site this is too much information (they shouldn't be seeing a message saying "this server is incorrectly configured" basically, and that's what this message says).

**Comment by zero-24** [comment]
I'm happy to accept a better suggestion. Maybe we show a different message in frontend and backend?

**Comment by brianteeman** [comment]
yes if possible to have a different message on the frontend that would be better

also please change him to them to make it gender neutral.

I will wait to here if the message can be split before commenting on the text itself

**Comment by mbabker** [comment]
The error message needs to be somewhat vague as to the cause of the issue still (the user just needs to know the upload failed, not everyone needs to know the technical reasoning for the failure), but we should be making it possible for site owners to get more information about why the issue happened.  Long and short, we need to start making more liberal use of our logging system and stop expecting that the only way to get information about an error state is through messages displayed to the user in the alert container.

So a message displayed in the UI should basically say not much more than "could not upload file because the MIME type could not be determined".  A supplemental `JLog::add()` call should be used to log additional details (upload of X file failed because the MIME type could not be determined).

**Comment by zero-24** [comment]
> I will wait to here if the message can be split

Yes it can as we have different language files for backend and frontend.

**Comment by brianteeman** [comment]
first attempt

### Frontend
+JLIB_MEDIA_ERROR_WARNINVALID_MIME="We were unable to upload this file."

### Backend
+JLIB_MEDIA_ERROR_WARNINVALID_MIME="We could not detect the mime type for this file. and for security purposes have not uploaded this file. Please contact your host and to check that mime_content_type or the fileinfo extension is enabled in php."

**Comment by zero-24** [comment]
done with the last commit.

**Comment by Quy** [comment]
I prefer not to not use `We were` and `We could` since we don't use `We` in the other strings.

How about?

`Unable to upload this file.`

`Unable to detect mime type for this file. For security purposes have not uploaded this file. Please contact your host and to check that mime_content_type or the fileinfo extension is enabled in PHP.`

**Comment by brianteeman** [comment]
i was trying to be more friendly - but if not then

Unable to upload this file.

Unable to detect the mime type for this file. For security purposes this file has not been uploaded . Please contact your host and check that the mime_content_type or the fileinfo extension is enabled in PHP.

**Comment by zero-24** [comment]
pushed thanks

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 1]
```diff
@@ -670,7 +670,8 @@ JLIB_MEDIA_ERROR_WARNFILETOOLARGE="This file is too large to upload."
 JLIB_MEDIA_ERROR_WARNFILETYPE="This file type is not supported."
 JLIB_MEDIA_ERROR_WARNIEXSS="Possible IE XSS Attack found."
 JLIB_MEDIA_ERROR_WARNINVALID_IMG="Not a valid image."
-JLIB_MEDIA_ERROR_WARNINVALID_MIME="Invalid mime type detected."
+; Do not sync that line with the frontend! This is expected to be different from the frontend string.
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: the next line instead of that line
please

**Comment by Quy** [comment]
> i was trying to be more friendly - but if not then

I totally understand. Just going for consistency sake.

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 2]
```diff
@@ -670,7 +670,8 @@ JLIB_MEDIA_ERROR_WARNFILETOOLARGE="This file is too large to upload."
...
+JLIB_MEDIA_ERROR_WARNINVALID_MIME="Unable to detect the mime type for this file. For security purposes this file has not been uploaded . Please contact your host and check that the mime_content_type or the fileinfo extension is enabled in PHP."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: extra space after uploaded

sorry

Comment (zero-24) [reply administrator/language/en-GB/en-GB.lib_joomla.ini]: fixed. np.

File: administrator/language/en-GB/en-GB.lib_joomla.ini (hunk 1 above)
Comment (zero-24) [reply administrator/language/en-GB/en-GB.lib_joomla.ini]: fixed

**Comment by Quy** [comment]
How about removing `this file` from the 1st sentence since it is mentioned (redundant) in the second sentence?

`Unable to detect the mime type. For security purposes this file has not been uploaded. Please contact your host and check that the mime_content_type or the fileinfo extension is enabled in PHP.`

**Comment by brianteeman** [comment]
Fine by me

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 11a71e9e7c170d9468857adb46ba330443200c23<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16246">issues.joomla.org/tracker/joomla-cms/16246</a>.</sub>

**Comment by brianteeman** [comment]
just had a thought and running for a train so cant test BUT is this message also displayed when you are trying to upload an file that is not in the allowed mime types that are saved in the com-media options? If so then this new message is not correct

**Comment by zero-24** [comment]
No than we have a differnent message that also include the detected mime type.
This message is only displayed if we could not detect a mimetype and this means that both methos are not enabled.

**Comment by brianteeman** [comment]
ok all good then

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 11a71e9e7c170d9468857adb46ba330443200c23<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16246">issues.joomla.org/tracker/joomla-cms/16246</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by mbabker** [comment]
I am still going to be insistent that the UI does not use an error message that says "you need X PHP extension/function enabled".  That comment is only beneficial if the only user who may see it is the site owner or can manage their server configuration.

**Comment by brianteeman** [comment]
@mbabker are the people who can see that message different to the people who can see the php update message?

**Comment by mbabker** [comment]
Depending on how you have the plugin configured or your site's ACL, yes, it could be a different group.

**Comment by rdeutz** [comment]
seems to me not really discussed how it should be, state set to "needs review"<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16246">issues.joomla.org/tracker/joomla-cms/16246</a>.</sub>

**Comment by zero-24** [comment]
I'm closing here as it looks like we can not find a consense here. Thanks.

**Pull request closed without merging**